plugins/pulp_rpm/plugins/importers/yum/config_validate.py

```diff
@@ -25,3 +46,69 @@ def validate(config):
             msg += failure_message + '\n'
         msg = msg.rstrip()  # remove the trailing \n
         return False, msg
+
+
+def _validate_allow_keys(repo, config, error_messages):
+    """
+    Validates that passed signature keys have proper lenth
+
+    :param repo: repository to validate the config for
+    :type  repo: pulp.plugins.model.Repository
+
+    :param config: configuration instance passed to the importer
+    :type  config: pulp.plugins.config.PluginCallConfiguration
+
+    :param error_messages: errors encountered during validation
+    :error_messages: list
+    """
+
+    allow_keys = config.get('allow_keys', set())
+    for key in allow_keys:
+        if len(key) != 8:
+            msg = _('Signature key %s should be 8 characters long') % key
+            error_messages.append(msg)
+
+
+def _validate_lazy_compatibility(repo, config, error_messages):
+    """
+    Validates lazy sync and signature policy compatibility.
+    To extract the signature from the package, we need this package been downloaded.
+
+    :param repo: repository to validate the config for
+    :type  repo: pulp.plugins.model.Repository
+
+    :param config: configuration instance passed to the importer
+    :type  config: pulp.plugins.config.PluginCallConfiguration
+
+    :param error_messages: errors encountered during validation
+    :error_messages: list
+    """
+
+    download_policy = config.get('download_policy', 'immediate')
+    allow_unsigned = config.get('allow_unsigned', True)
+    allow_keys = config.get('allow_keys', set())
+    if download_policy in ('on_demand', 'background') and (not allow_unsigned or allow_keys):
+        msg = _('%s download policy and signature check are not compatible') % download_policy
+        error_messages.append(msg)
+
+
+def _validate_allow_unsigned(repo, config, error_messages):
+    """
+    Validates that allow_unsogned cannot be enabled when there is content in the repo.
```

Comment:
I'm not looking at this long enough to also be confused. Don't we want the reverse of this actually?

Allowing more valid rpms should still keep Pulp's guarantees to the user valid. Enabling the allow_unsigned does that so we should allow it to be enabled when there is content in the repo. We should prevent the disabling of `allow_unsigned` because it's disabling which would cause the units in the repo to be inconsistent with the most recent config update. Does that make sense?